public/app/features/query/state/DashboardQueryRunner/UnifiedAlertStatesWorker.ts

```diff
@@ -54,53 +59,53 @@ export class UnifiedAlertStatesWorker implements DashboardQueryRunnerWorker {
     }
 
     const { dashboard } = options;
-    return from(
-      getBackendSrv().get(
-        '/api/prometheus/grafana/api/v1/rules',
-        {
-          dashboard_uid: dashboard.uid,
-        },
-        `dashboard-query-runner-unified-alert-states-${dashboard.id}`
-      )
-    ).pipe(
-      map((result: PromRulesResponse) => {
-        if (result.status === 'success') {
-          this.hasAlertRules[dashboard.uid] = false;
-          const panelIdToAlertState: Record<number, AlertStateInfo> = {};
-          result.data.groups.forEach((group) =>
-            group.rules.forEach((rule) => {
-              if (isAlertingRule(rule) && rule.annotations && rule.annotations[Annotation.panelID]) {
-                this.hasAlertRules[dashboard.uid] = true;
-                const panelId = Number(rule.annotations[Annotation.panelID]);
-                const state = promAlertStateToAlertState(rule.state);
+    const fetchData: () => Promise<RuleNamespace[]> = async () => {
+      const promRules = dispatch(
```

Comment:
It doesn't.  
This await for promRules is used to wait for the Promise returned by the dispatch function to resolve. Once it's resolved, .data is accessed from the resolved value.

If await is not used, promRules is a Promise, and trying to access .data directly from a Promise results in undefined because the Promise has not yet resolved.

Reply:
updated!